src/gui/tray/UserModel.cpp

```diff
@@ -58,52 +59,43 @@ User::User(AccountStatePtr &account, const bool &isCurrent, QObject *parent)
     connect(_activityModel, &ActivityListModel::sendNotificationRequest, this, &User::slotSendNotificationRequest);
 }
 
-void User::slotBuildNotificationDisplay(const ActivityList &list)
+void User::showDesktopNotification(const QString &title, const QString &message)
 {
-    // Whether a new notification was added to the list
-    bool newNotificationShown = false;
+    ConfigFile cfg;
+    if (!cfg.optionalServerNotifications() || !isDesktopNotificationsAllowed()) {
+        return;
+    }
 
+    // after one hour, clear the gui log notification store
+    if (_guiLogTimer.elapsed() > 60 * 60 * 1000) {
+        _notificationCache.clear();
+    }
+
+    const Notification notification { title, message };
+    const auto isNotificationCached = _notificationCache.contains(notification);
```

Comment:
@FlexW This variable seem redundant to me :)

Reply:
Yes true.